cl/search/exception.py

```diff
@@ -47,7 +62,23 @@ class ElasticBadRequestError(APIException):
     """Exception for handling ES query parsing errors."""
 
     status_code = HTTPStatus.BAD_REQUEST
-    default_detail = (
-        "Elasticsearch Bad request error. Please review your query."
-    )
+    default_detail = default_bad_request_error_msg
     default_code = "bad_request"
+
+
+class UnbalancedParenthesesQueryAPIError(ElasticBadRequestError):
+    """APIException for handling unbalanced parentheses in ES queries."""
+
+    default_detail = unbalanced_parentheses_error_msg
+
+
+class UnbalancedQuotesQueryAPIError(ElasticBadRequestError):
+    """APIException for handling unbalanced quotes in ES queries."""
+
+    default_detail = unbalanced_quotes_error_msg
+
+
+class BadProximityQueryAPIError(ElasticBadRequestError):
+    """APIException for handling bad proximity values in ES queries."""
+
+    default_detail = bad_proximity_query_error_msg
```

Comment:
Is there a way not to do these? They seem pretty clunky?

Reply:
Yeah, the only thing that changes is the `detail` parameter. I've removed these exceptions and instead use `ElasticBadRequestError`, just passing their specific message.

Thanks for the suggestion.

Reply:
Thanks for the tweak! Much better now!